lib/buffer.js

```diff
@@ -852,6 +852,12 @@ function checkOffset(offset, ext, length) {
     throw new RangeError('Index out of range');
 }
 
+function checkByteLength(byteLength) {
+  if (typeof byteLength !== 'number' || byteLength <= 0)
+    throw new TypeError('"byteLength" must be a positive number ' +
```

Comment:
I think separate errors would be better, because a `TypeError` for `byteLength <= 0` doesn't look right IMO.

According to the docs:

> `byteLength` {Integer} How many bytes to read. Must satisfy: `0 < byteLength <= 6`

and 

> Supports up to 48 bits of accuracy.

So probably check `byteLength > 6` too?

Also a doc update would be great :D